rpc/l1.go

```diff
@@ -0,0 +1,112 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/jsonrpc"
+	"github.com/ethereum/go-ethereum/common"
+	"golang.org/x/crypto/sha3"
+)
+
+var ErrL1ClientNotFound = jsonrpc.Err(jsonrpc.InternalError, fmt.Errorf("L1 client not found, cannot serve starknet_getMessage"))
+
+type LogMessageToL2 struct {
+	FromAddress *common.Address
+	ToAddress   *common.Address
+	Nonce       *big.Int
+	Selector    *big.Int
+	Payload     []*big.Int
+	Fee         *big.Int
+}
+
+// HashMessage calculates the message hash following the Keccak256 hash method
+func (l *LogMessageToL2) HashMessage() *common.Hash {
+	hash := sha3.NewLegacyKeccak256()
+
+	// Padding for Ethereum address to 32 bytes
+	hash.Write(make([]byte, 12)) //nolint:mnd
+	hash.Write(l.FromAddress.Bytes())
+	hash.Write(l.ToAddress.Bytes())
+	hash.Write(l.Nonce.Bytes())
+	hash.Write(l.Selector.Bytes())
+
+	// Padding for payload length (u64)
+	hash.Write(make([]byte, 24))     //nolint:mnd
+	payloadLength := make([]byte, 8) //nolint:mnd
+	big.NewInt(int64(len(l.Payload))).FillBytes(payloadLength)
+	hash.Write(payloadLength)
+
+	for _, elem := range l.Payload {
+		hash.Write(elem.Bytes())
+	}
+	tmp := common.BytesToHash(hash.Sum(nil))
+	return &tmp
+}
+
+type MsgStatus struct {
+	L1HandlerHash  *felt.Felt
+	FinalityStatus TxnStatus
+	FailureReason  string
+}
```

Comment:
Do you think we need to have `json` tags here? The field names do not match the names required in the spec. Also the failure reason should be omitted if it is empty.

```suggestion
type MsgStatus struct {
	L1HandlerHash  *felt.Felt `json:"transaction_hash"`
	FinalityStatus TxnStatus  `json:"finality_status"`
	FailureReason  string     `json:"failure_reason,omitempty"`
}
```

Reply:
Yes, thank you!